app/parsers/bulkrax/application_parser.rb

```diff
@@ -37,15 +38,94 @@ def records(opts = {})
       raise 'must be defined'
     end
 
-    def import_fields
-      raise 'must be defined'
+    def import_file_path
+      @import_file_path ||= self.parser_fields['import_file_path']
+    end
+
+    def create_collections
+      raise 'must be defined' if importer?
+    end
+
+    def create_works
+      raise 'must be defined' if importer?
+    end
+
+    # Optional, only used by certain parsers
+    # Other parsers should override with a custom or empty method
+    # Will be skipped unless the record is a Hash
+    def create_parent_child_relationships
+      parents = setup_parents
```

Comment:
see comment below on why I think these 2 lines can be deleted (since [].each does nothing and doesn't need to be skipped)